lib/ransack/adapters/active_record/base.rb

```diff
@@ -45,7 +45,7 @@ def ransackable_attributes(auth_object = nil)
         # For overriding with a whitelist array of strings.
         #
         def ransackable_associations(auth_object = nil)
-          reflect_on_all_associations.map { |a| a.name.to_s }
+          @ransackable_associations ||= reflect_on_all_associations.map { |a| a.name.to_s }
```

Comment:
You'll want to be careful about memoization in these -- that's why they weren't to begin with. Technically, new values could be added by code at runtime -- they would need to invalidate the cache.

Reply:
It appears that we can use the `to_prepare` Rails callback to expire the cache, though that would require that we do one of these:
- instead of using instance variables, cache the values in a global object
- cache the values in instance variables, but via a module that would hook into `to_prepare`

Reply:
It appears that we don't currently have tests to verify that Rails code reloading works with Ransack. IIRC ActiveAdmin has tests of that type, but they require a lot of setup code.

Reply:
Actually `to_prepare` is unnecessary since these methods are defined on the models themselves. In the Paper Trail case, adding `has_paper_trail` to the model would unload & reload the model, so `ransackable_associations` is no longer cached.

AFAICT there aren't any scenarios where this could cause a bug.